util/crypto/src/secp/generator.rs

```diff
@@ -13,16 +13,23 @@ impl Generator {
     }
 
     pub fn random_privkey(&self) -> Privkey {
-        let mut rng = rand::thread_rng();
-        key::SecretKey::new(&SECP256K1, &mut rng).into()
+        let mut random_slice: [u8; secp256k1::constants::SECRET_KEY_SIZE] = rand::random();
```

Comment:
secp256k1 re-exports dep crate after version 0.11.5:
https://github.com/rust-bitcoin/rust-secp256k1/blob/master/CHANGELOG.md#0115---2018-11-09

We can use this feature directly:
```rust
SecretKey::new(&mut secp256k1::rand::thread_rng()).into()
```

Reply:
@quake Why do we fork it? just because of `rand` 0.4 0.5 0.6 coexistence will conflict.

By writing this, I can guarantee that we are using the latest secp256 without affecting the use of the latest `rand` elsewhere.

I don't think we need to fork, nor do we need to force downgrade the `rand` crate.

Reply:
not forking, just enable the `rand` feature in Cargo.toml:
```
secp256k1 = { version = "0.12.2", optional = true, features = ["rand"] }
```

Reply:
Secp256k1 exports a version of 0.4, but we don't really need this feature, so I don't use it.

I remember that there was a problem with `rand` crate multi-version conflicts in history. In order to reduce this unnecessary trouble, I think really does not need to use this feature

Reply:
Yes, secp256k1 has a version conflict issue before 0.11.5 and therefore it exports dep crate to resolve this issue:

https://github.com/rust-bitcoin/rust-secp256k1/pull/80

Reply:
We use `rand` 0.6 on `ckb-protocol`、`ckb-shared`、`bloom-filters` and lot of crate.

Enable secp256k1 `rand` feature doesn't give us any benefit other than letting us compile more than one library.